pegasus/sites.v3/code.org/public/teacher-dashboard/section_assessments.haml

```diff
@@ -16,6 +17,58 @@ content-type: text/ng-template
 %br/
 
 :ruby
+  surveys_filtered = "surveyLevels | orderBy:orderSurveyAnswers | filter:(stageFilterSurveyAnswers && {stage: stageFilterSurveyAnswers}):true"
+
+%h1{'ng-show' => 'surveys.length', style: "margin-top: 0px"}= I18n.t('dashboard_surveys_answers')
+
+%div{style: "padding-left:20px", 'ng-show' => 'surveys.length'}
+  %table{style: "width:950px; table-layout:fixed;", 'ng-show' => 'surveys.length', "ng-form" => "allForm"}
+    %colgroup
+      %col{width: "250"}/
+      %col{width: "75"}/
+      %col{width: "250"}/
+      %col{width: "225"}/
+      %col{width: "75"}/
+
+    %tr
+      %th.manage-th{colspan: 5}
+        %div{'ng-show' => 'surveyStages && surveyStages.length > 1', style: 'float: left; line-height: 35px'}
+          = I18n.t('dashboard_filter_by_stage')
+          %select{'ng-model' => 'stageFilterSurveyAnswers', "ng-options" => "stage for stage in surveyStages", "ng-show" => 'surveyStages'}
+            %option{value: ""}= I18n.t('dashboard_filter_all')
+        %button.btn.btn-white{'ng-csv' => surveys_filtered,
+                filename: "survey_levels.csv",
+                'csv-header' => "['Name', 'Stage', 'Puzzle', 'Question', 'Response', 'Correctness']",
+                'csv-column-order' => "['student.name', 'stage', 'puzzle', 'question', 'studentResult', 'count']",
+                style: 'float: right'}
+          %i.fa.fa-download
+          = I18n.t('dashboard_download_csv')
+
+    %tr
+      %th.manage-th
+        %a{href: "", "ng-click" => "orderSurveyAnswers = 'stage'"}= I18n.t('dashboard_stage')
+      %th.manage-th
+        %a{href: "", "ng-click" => "orderSurveyAnswers = 'puzzle'"}= I18n.t('dashboard_puzzle')
+      %th.manage-th
+        %a{href: "", "ng-click" => "orderSurveyAnswers = 'question'"}= I18n.t('dashboard_question')
+      %th.manage-th
+        %a{href: "", "ng-click" => "orderSurveyAnswers = ['!studentResult', 'studentResult']"}= I18n.t('dashboard_response')
+      %th.manage-th
+        %a{href: "", "ng-click" => "orderSurveyAnswers = 'count'"}= I18n.t('dashboard_percent')
+
+    %tr{"ng-repeat" => "level in #{surveys_filtered}", "ng-form" => "form"}
+      %td
+        {{level.stage}}
+      %td
+        {{level.puzzle}}
+      %td
+        {{level.questionText}}
+      %td
+        {{level.resultText}}
+      %td
+        {{level.percent}}
```

Comment:
I can't think of any technical blockers, it seems more just a rewrite of a whole bunch of interactive UI (mostly tables) across all of teacher dashboard.